graylog2-web-interface/src/components/search/Decorator.jsx

```diff
@@ -63,22 +66,32 @@ const Decorator = React.createClass({
 
     return Object.assign({}, config, resolvedConfig);
   },
+  _formatActionsMenu() {
+    const permissions = this.state.currentUser.permissions;
+    const decorator = this.props.decorator;
+    const editPermission = this.isPermitted(permissions, `decorators:edit:${decorator.stream}`);
+    return (
+      <DropdownButton id={`decorator-${decorator._id}-actions`} bsStyle="default" bsSize="xsmall" title="Actions" pullRight>
+        <MenuItem onSelect={this._handleEditClick} disabled={!editPermission}>Edit</MenuItem>
+        <MenuItem divider/>
+        <MenuItem onSelect={this._handleDeleteClick} disabled={!editPermission}>Delete</MenuItem>
+      </DropdownButton>
+    );
+  },
   render() {
-    if (!this.state.types) {
+    if (!this.state.types || !this.state.currentUser) {
       return <Spinner />;
     }
     const decorator = this.props.decorator;
     const config = this._resolveConfigurationIds(decorator.config);
     const decoratorType = this.state.types[decorator.type] || this._decoratorTypeNotPresent();
+
+    const decoratorActionsMenu = this._formatActionsMenu();
     return (
       <span className={DecoratorStyles.fullWidth}>
         <div className={DecoratorStyles.decoratorBox}>
           <h6 className={DecoratorStyles.decoratorType}>{decoratorType.name}</h6>
-          <DropdownButton id={`decorator-${decorator._id}-actions`} bsStyle="default" bsSize="xsmall" title="Actions" pullRight>
-            <MenuItem onSelect={this._handleEditClick}>Edit</MenuItem>
-            <MenuItem divider/>
-            <MenuItem onSelect={this._handleDeleteClick}>Delete</MenuItem>
-          </DropdownButton>
+          {decoratorActionsMenu}
```

Comment:
My general rule of thumb for hiding functionality vs. disabling it is that:

  * hiding functionality which the user will never be able to use (i.e. requiring admin privileges)
  * disabling functionality which is not available in this context, but might be at some other point

This way it is transparent for the user (although in this case we could add some helpful tooltips) that the functionality exists, but is not usable (yet) and the layout stays the same.

We do not do that consistently, but that's not because the way we do it now is the right one in most places, but because we do not have a convention for this yet. :)